src/agreement.rs

```diff
@@ -94,8 +94,25 @@ pub struct Algorithm {
     // `ec::AgreementAlgorithmImpl` isn't public.
     #[doc(hidden)]
     pub i: ec::AgreementAlgorithmImpl,
+    #[doc(hidden)]
+    pub id: AlgorithmID,
+}
+
+/// Key agreement algorithm types.
+#[allow(non_camel_case_types)]
+#[derive(PartialEq, Eq)]
+pub enum AlgorithmID {
+    ECDH_P256,
+    ECDH_P384,
+    X25519,
 }
```

Comment:
`CurveID` already exists, so we don't need this. I'll remove this in favor of an approach using `CurveID`.